robowflex_library/include/robowflex_library/builder.h

```diff
@@ -0,0 +1,234 @@
+         *  \param[in] joints The values of the joints to set.
+         */
+        void setGoalConfiguration(const std::vector<double> &joints);
+
+        /** \brief Set the goal configuration from a robot state.
+         *  \param[in] state The robot state to set. Usually from robowflex::Robot::getScratchState().
+         */
+        void setGoalConfiguration(const robot_state::RobotStatePtr &state);
+
+        /** \brief Set a goal region for an end-effector \a ee_name.
+         *  Sets the position constraint from \a geometry at a pose \a pose, and the orientation constraint
+         *  from \a orientation and XYZ Euler angle tolerances \a tolerances.
+         *  \param[in] ee_name The name of the end-effector link.
+         *  \param[in] base_name The frame of pose and orientation.
+         *  \param[in] pose The pose of \a geometry in \a base_frame.
+         *  \param[in] geometry The geometry describing the position constraint.
+         *  \param[in] orientation The desired orientation.
+         *  \param[in] tolerances XYZ Euler angle tolerances about orientation.
+         */
+        void setGoalRegion(const std::string &ee_name, const std::string &base_name,
+                           const Eigen::Affine3d &pose, const GeometryConstPtr &geometry,
+                           const Eigen::Quaterniond &orientation, const Eigen::Vector3d &tolerances);
+
+        /** \brief Tiles a \a region around a \a pose in \a base_name for the end-effector \a ee_name. The \a
+         * region is placed at \a offset from \a pose, and \a n copies are placed evenly rotated about \a
+         * axis.
+         *  \param[in] ee_name The name of the end-effector link.
+         *  \param[in] base_name The frame of pose and orientation.
+         *  \param[in] pose The pose of the frame to be rotated about.
+         *  \param[in] geometry The geometry describing the position constraint.
+         *  \param[in] orientation The desired orientation.
+         *  \param[in] tolerances XYZ Euler angle tolerances about orientation.
+         *  \param[in] offset Offset of the goal region from \a pose.
+         *  \param[in] axis Axis to rotation the goal region about in \a pose.
+         *  \param[in] n Number of rotations (evenly divided around the circle).
+         */
+        void addGoalRotaryTile(const std::string &ee_name, const std::string &base_name,
+                               const Eigen::Affine3d &pose, const GeometryConstPtr &geometry,
+                               const Eigen::Quaterniond &orientation, const Eigen::Vector3d &tolerances,
+                               const Eigen::Affine3d &offset, const Eigen::Vector3d &axis, unsigned int n);
+
+        /** \brief Adds a set of regions to grasp a cylinder from the side.
+         *  \param[in] ee_name The name of the end-effector link.
+         *  \param[in] base_name The frame of pose and orientation.
```

Comment:
The description for `base_name` (and on other funcs too) is a little vaguely worded when I first read it, mostly because "The frame of pose" comes right before "The pose of the frame".  Can you either:
* put `\a` in front of pose and orientation to show that you're referencing other variables.
* add a few more descriptor words, i.e. The frame of reference for pose and orientation, etc?